warrior/Framework/Utils/data_Utils.py

```diff
@@ -973,6 +972,84 @@ def validate():
     return result, value
 
 
+def verify_arith_exp(expression, expected, comparison='eq'):
+    """ Verify the output of the arithmetic expression matches the expected(float comparison)
+        Note : Binary floating-point arithmetic holds many surprises.
+        Please refer to link, https://docs.python.org/2/tutorial/floatingpoint.html
+        This Keyword inherits errors in Python float operations.
+        :Arguments:
+            1. expression: Arithmetic expression to be compared with expected.
+                This can have env & data_repo values embedded in it.
+                    Ex. expression: "10+${ENV.x}-${REPO.y}*10"
+                Expression will be evaluated based on python operator precedence
+            2. expected: Value to be compared with the expression output
+                This can be a env or data_repo or any numeral value.
+            3. comparison: Type of comparison(eq/ne/gt/ge/lt/le)
+                eq - check if both are same(equal)
+                ne - check if both are not same(not equal)
+                gt - check if expression output is greater than expected
+                ge - check if expression output is greater than or equal to expected
+                lt - check if expression output is lesser than expected
+                le - check if expression output is lesser than or equal to expected
+        :Returns:
+            1. status(boolean)
+    """
+
+    status = True
+    # Substitute env values in the expression & expected
+    expression = sub_from_env_var(expression)
+    expected = sub_from_env_var(expected)
+    # Substitute data_repo values in the expression & expected
+    expression = sub_from_data_repo(expression)
+    expected = sub_from_data_repo(expected)
+    #Safe expression pattern from user.
+    expression_pattern = "^[0-9^%+\-*\/\(\)]*$"
+    #Any possible unsafe expression from user avoids passing through EVAL.
+    if not re.search(expression_pattern, expression):
+        print_error("Unable to evaluate the expression '{}' provided. "
```

Comment:
Please change this error message to explain it is an illegal expression...

Reply:
There is a possibility of unresolved repo & ENV values.
Added the second possibility as illegal expression,